### PR TITLE
Update language around client version compatibility

### PR DESCRIPTION
**Pull request by hdhalter**
### Description
Addresses community feedback from Slack. 

### Issues Resolved
See https://opensearch.slack.com/archives/C052GAV1MQF/p1712689014138299

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by nhtruong — APPROVED** [review]
Looks good!

File: _clients/index.md [hunk 1]
```diff
@@ -39,18 +42,14 @@ OpenSearch provides clients for the following programming languages and platform
...
-For a client compatibility matrix, see the COMPATIBILITY.md file in the client's repository.
-{: .note}
-
-The OpenSearch Java high-level REST client will be deprecated starting with OpenSearch version 3.0.0 and will be removed in a future release. We recommend switching to the [Java client]({{site.url}}{{site.baseurl}}/clients/java/) instead.
-{: .warning}
-
 ## Legacy clients
 
-Most clients that work with Elasticsearch OSS 7.10.2 *should* work with OpenSearch, but the latest versions of those clients might include license or version checks that artificially break compatibility. This page includes recommendations around which versions of those clients to use for best compatibility with OpenSearch.
+Most clients that work with ElasticSearch OSS 7.10.2 *should* work with OpenSearch 1.x, but the latest versions of those clients might include license or version checks that artificially break compatibility. The table that follows includes recommendations for which client versions to use for best compatibility with OpenSearch 1.x. From OpenSearch 2.0 and later, no ElasticSearch clients are fully compatible with OpenSearch.
```
Comment (Naarcha-AWS) [inline _clients/index.md:47]:
```suggestion
Most clients that work with Elasticsearch OSS 7.10.2 *should* work with OpenSearch 1.x, but the latest versions of those clients might include license or version checks that artificially break compatibility. The table that follows includes recommendations for which client versions to use for best compatibility with OpenSearch 1.x. From OpenSearch 2.0 and later, no Elasticsearch clients are fully compatible with OpenSearch.
```

File: _clients/index.md [hunk 2]
```diff
@@ -39,18 +42,14 @@ OpenSearch provides clients for the following programming languages and platform
...
+
+While OpenSearch and ElasticSearch share a majority of core features, mixing and matching the client and server has a very high risk of errors and unexpected results. The issues will continue to get worse over time as OpenSearch and ElasticSearch diverge even further from each other. So, even if your ElasticSearch client still works with your OpenSearch cluster, we highly recommend using OpenSearch clients for OpenSearch clusters.
```
Comment (Naarcha-AWS) [inline _clients/index.md:49]:
```suggestion
While OpenSearch and Elasticsearch share a majority of core features, mixing and matching the client and server has a very high risk of errors and unexpected results. The issues will continue to get worse over time as OpenSearch and Elasticsearch diverge even further from each other. So, even if your Elasticsearch client still works with your OpenSearch cluster, we highly recommend using OpenSearch clients for OpenSearch clusters.
```

**Review by vagimeli — APPROVED** [review]
Please see the suggested edits to tighten or clarify writing. To avoid duplicative efforts, has this been scheduled for editorial review?

File: _clients/index.md (hunk 2 above)
Comment (vagimeli) [inline _clients/index.md:49]:
```suggestion
While OpenSearch and Elasticsearch share several core features, mixing and matching the client and server has a high risk of errors and unexpected results. As OpenSearch and Elasticsearch continue to diverge, such risks may increase. Although your Elasticsearch client may continue working with your OpenSearch cluster, using OpenSearch clients for OpenSearch clusters is recommended.
```

**Review by vagimeli — APPROVED** [review]

File: _clients/index.md [hunk 3]
```diff
@@ -39,18 +42,14 @@ OpenSearch provides clients for the following programming languages and platform
...
-For a client compatibility matrix, see the COMPATIBILITY.md file in the client's repository.
-{: .note}
-
-The OpenSearch Java high-level REST client will be deprecated starting with OpenSearch version 3.0.0 and will be removed in a future release. We recommend switching to the [Java client]({{site.url}}{{site.baseurl}}/clients/java/) instead.
-{: .warning}
-
 ## Legacy clients
 
-Most clients that work with Elasticsearch OSS 7.10.2 *should* work with OpenSearch, but the latest versions of those clients might include license or version checks that artificially break compatibility. This page includes recommendations around which versions of those clients to use for best compatibility with OpenSearch.
+Most clients that work with Elasticsearch OSS 7.10.2 *should* work with OpenSearch 1.x, but the latest versions of those clients might include license or version checks that artificially break compatibility. The table that follows includes recommendations for which client versions to use for best compatibility with OpenSearch 1.x. From OpenSearch 2.0 and later, no Elasticsearch clients are fully compatible with OpenSearch.
```
Comment (hdhalter) [inline _clients/index.md:47]:
```suggestion
Clients that work with Elasticsearch OSS 7.10.2 should work with OpenSearch 1.x. The latest versions of those clients, however, might include license or version checks that artificially break compatibility. The following table provides recommendations for which client versions to use for best compatibility with OpenSearch 1.x. For OpenSearch 2.0 and later, no Elasticsearch clients are fully compatible with OpenSearch.
```

**Review by hdhalter — COMMENTED** [review]
@vagimeli - Thank you for the review. I may have overwritten one of your comments by accepting a previous one. Can you please review this one comment and accept if this looks good? Thanks!

**Pull request merged**